# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-am-build libxext-build libxrender-build"

PKG_CONFIG_PATH=/x11/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure (ac-2.63; am-1.11)
aclocal; autoheader; autoconf; automake -a -c
PKG_CONFIG_PATH=/x11/lib/pkgconfig LDFLAGS=-Wl,-rpath,/x11/lib ./configure --prefix=/x11

# build
make

# install
install -d $DESTDIR/x11/bin
install xdpyinfo $DESTDIR/x11/bin

# build.pc
install -d $DESTDIR/x11/share/pkgconfig
cat > $DESTDIR/x11/share/pkgconfig/xdpyinfo-build.pc << EOF
Name: xdpyinfo
Version: 1.1.0
Description: Display information utility for X
Requires: $REQUIRES

exec=\\
/x11/bin/xdpyinfo
EOF

exit

# patch
--- xdpyinfo-1.1.0.orig/configure.ac
+++ xdpyinfo-1.1.0/configure.ac
@@ -38,7 +38,7 @@
 XORG_DEFAULT_OPTIONS
 
 # Checks for pkg-config packages
-PKG_CHECK_MODULES(XDPYINFO, xext x11 xtst)
+PKG_CHECK_MODULES(XDPYINFO, xext x11)
 AC_SUBST(XDPYINFO_CFLAGS)
 AC_SUBST(XDPYINFO_LIBS)
 
@@ -146,11 +146,17 @@
 	echo "without xprint"
 fi
 
-PKG_CHECK_MODULES(DPY_XTST, xtst, 
-	[SAVE_CPPFLAGS="$CPPFLAGS"
-	CPPFLAGS="$CPPFLAGS $DPY_XTST_CFLAGS $DPY_X11_CFLAGS"
-	AC_CHECK_HEADERS([X11/extensions/record.h],,,[#include <X11/Xlib.h>])
-	CPPFLAGS="$SAVE_CPPFLAGS"],[echo "not found"])
+AC_ARG_WITH(xtst, AC_HELP_STRING([--without-xtst],[Disable xtst support.]), 
+		[USE_XTST="$withval"], [USE_XTST="yes"])
+if test "x$USE_XTST" != "xno" ; then
+	PKG_CHECK_MODULES(DPY_XTST, xtst, 
+		[SAVE_CPPFLAGS="$CPPFLAGS"
+		CPPFLAGS="$CPPFLAGS $DPY_XTST_CFLAGS $DPY_X11_CFLAGS"
+		AC_CHECK_HEADERS([X11/extensions/record.h],,,[#include <X11/Xlib.h>])
+		CPPFLAGS="$SAVE_CPPFLAGS"],[echo "not found"])
+else
+	echo "without xtst"
+fi
 
 dnl Allow checking code with lint, sparse, etc.
 XORG_WITH_LINT
--- xdpyinfo-1.1.0.orig/xdpyinfo.c
+++ xdpyinfo-1.1.0/xdpyinfo.c
@@ -84,6 +84,10 @@
 #  define INCLUDE_XPRINT_SUPPORT
 # endif
 
+# if HAVE_X11_EXTENSIONS_RECORD_H
+#  define XTEST_AND_RECORD
+# endif
+
 #endif
 
 #include <X11/Xlib.h>
@@ -91,11 +95,15 @@
 #ifdef MULTIBUFFER
 #include <X11/extensions/multibuf.h>
 #endif
+#ifdef XTEST_AND_RECORD
 #include <X11/extensions/XTest.h>
+#endif
 #include <X11/extensions/sync.h>
 #include <X11/Xproto.h>
 #include <X11/extensions/Xdbe.h>
+#ifdef XTEST_AND_RECORD
 #include <X11/extensions/record.h>
+#endif
 #include <X11/extensions/shape.h>
 #ifdef MITSHM
 #include <X11/extensions/XShm.h>
@@ -693,6 +701,7 @@
 } /* end print_multibuf_info */
 #endif
 
+#ifdef XTEST_AND_RECORD
 static int
 print_xtest_info(Display *dpy, char *extname)
 {
@@ -703,6 +712,7 @@
     print_standard_extension_info(dpy, extname, majorrev, minorrev);
     return 1;
 }
+#endif
 
 static int
 print_sync_info(Display *dpy, char *extname)
@@ -1019,6 +1029,7 @@
     return 1;
 }
 
+#ifdef XTEST_AND_RECORD
 static int
 print_record_info(Display *dpy, char *extname)
 {
@@ -1029,6 +1040,7 @@
     print_standard_extension_info(dpy, extname, majorrev, minorrev);
     return 1;
 }
+#endif
 
 #ifdef XINPUT
 static int
@@ -1482,9 +1494,13 @@
 #ifdef XF86MISC
     {XF86MISCNAME, print_XF86Misc_info, False},
 #endif /* XF86MISC */
+#ifdef XTEST_AND_RECORD
     {XTestExtensionName, print_xtest_info, False},
+#endif /* XTEST_AND_RECORD */
     {"DOUBLE-BUFFER", print_dbe_info, False},
+#ifdef XTEST_AND_RECORD
     {"RECORD", print_record_info, False},
+#endif /* XTEST_AND_RECORD */
 #ifdef XINPUT
     {INAME, print_xinput_info, False},
 #endif

# source
https://x.org/releases/individual/app/xdpyinfo-1.1.0.tar.gz
